core/adapters/condition_test.go

```diff
@@ -0,0 +1,323 @@
+package adapters_test
+
+import (
+	"testing"
+
+	"github.com/smartcontractkit/chainlink/core/adapters"
+	"github.com/smartcontractkit/chainlink/core/internal/cltest"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCondition_Perform(t *testing.T) {
+	tests := []struct {
+		name       string
+		input      string
```

Comment:
This could be something other than a string. Although it looks like it would work now even without tests that are strings, let's add some tests for ints/floats/bools to avoid regressions.